flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -280,6 +317,54 @@
 	 */
 	void sqlUpdate(String stmt, QueryConfig config);
 
+	/**
+	 * Gets the current default catalog name of the current session.
+	 *
+	 * @return The current default catalog name that is used for the path resolution.
+	 * @see TableEnvironment#useCatalog(String)
+	 */
+	String getCurrentCatalog();
+
+	/**
+	 * Sets the current catalog to the given value. It also sets the default
+	 * database to the catalog's default one. To assign both catalog and database explicitly
+	 * see {@link TableEnvironment#useDatabase(String, String)}.
+	 *
+	 * <p>This is used during the resolution of object paths. The default path is constructed as
+	 * {@code [current-catalog].[current-database]}. During the resolution, first we try to look for
+	 * {@code [default-path].[object-path]} if no object is found we assume the object path is a fully
+	 * qualified one and we look under {@code [object-path]}.
+	 *
+	 * @param catalogName The name of the catalog to set as the current default catalog.
+	 * @throws CatalogException thrown if a catalog with given name could not be set as the default one
+	 */
+	void useCatalog(String catalogName);
```

Comment:
nit: can we add the exception (even though it's runtime exception) to signature to inform users this may throw exception, we've done that for catalog APIs

Reply:
This is against majority (if not all) code style guidelines. The `throws` cause should declare only those exception that we expect the user to handle.

BTW, me & @twalthr had a chat few times before and we fill we should revisit the exception structure in the catalog API anyway.

Reply:
Adding CatalogException to signature is actually suggested by Timo at https://github.com/apache/flink/pull/8007#discussion_r275425557 .

I agree we should revisit the exceptions as I too find it inconsistent and problematic

Reply:
My comment can be a bit misunderstanding. Yeah, my point is we just need to be consistent with how it's exposed. An example is that now, contrary to `useCatalog()`, `Catalog`'s APIs not only document `CatalogException` in javadoc but also `throws CatalogException` in their signatures.

Seems that we all agree upon the javadoc part. What do you guys think of the signature part? If we all feel it's not necessary to expose it to users in the API, I feel it's better to remove `throws CatalogException` from `Catalog`'s API signatures to be consistent.

Reply:
I don't have a strong opinion here. Personally, I would add it to the signature if it is an important interface to highlight the exception. I mean there is a reason why Java also support declaring runtime exceptions explicitly. But I would definitely not bubble the runtime exception up in the stack of signatures.

Reply:
But I think the root cause here is that `CatalogException` is not a checked exception. I suggested it to handle communication failures similar to a `IOException`.